### ISSUE TITLE
Graduation of EventBridge Feature

### ISSUE THREAD

**Issue by sedefsavas** [kubernetes-sigs/cluster-api-provider-aws#3414]
This issue is to track graduation of the Eventbridge feature, which currently only watches instance state change events. 

There are other type of CloudWatch events that can be watched and acted upon such as ASG Scale in events and Spot Instance Termination Notices. 

Some use cases for these:

- In the current AWSMachinePool implementation, we don't have draining logic and with the use of lifecyclehooks, we can act on the instance termination events and drain the instances during rollout or scale-in. (see [graceful shutdown of MachinePools ](https://github.com/kubernetes-sigs/cluster-api-provider-aws/pull/3256)for details)
- In the near future we are planning to add support for[ spot instances with AWSMachinePool](https://github.com/kubernetes-sigs/cluster-api-provider-aws/issues/2523) and termination notices could give us enough time to drain spot instances before they shut down (notices give about 2 minutes in some scenarios)

In the scope of this effort, the existing implementation should be modified to accommodate different types of events and should be flexible enough to be used by several controllers. Also, for cases where events need to be exposed to the controllers (this will be the case probably for AWSMachinePool draining), some helper functions should make read/delete interaction with events possible.

**Comment by richardcase** [comment]
@Ankitasw - shall we look at the geduation in the new year?

**Comment by Ankitasw** [comment]
Yes, i started it a while ago, but got busy in focussing on E2E, my focus after E2E things would be this.

**Comment by fiunchinho** [comment]
hey @Ankitasw , is there a way to track the progress on this? We are very interested on https://github.com/kubernetes-sigs/cluster-api-provider-aws/issues/2574 , but it seems to be blocked by this issue. Thank you!

**Comment by Ankitasw** [comment]
This is in our radar, would try to get this prioritized soon.

**Comment by fiunchinho** [comment]
hey @Ankitasw , just checking in :slightly_smiling_face: . Any progress to share?
